airbyte-webapp/src/components/connectorBuilder/Builder/InputsView.tsx

```diff
@@ -0,0 +1,303 @@
+  };
+}
+
+function inputInEditingToFormInput({
+  type,
+  showDefaultValueField,
+  isNew,
+  ...values
+}: InputInEditing): BuilderFormInput {
+  return {
+    ...values,
+    definition: {
+      ...values.definition,
+      type: type === "enum" ? "string" : type === "unknown" ? values.definition.type : type,
+      // only respect the enum values if the user explicitly selected enum as type
+      enum: type === "enum" && values.definition.enum?.length ? values.definition.enum : undefined,
+      default: showDefaultValueField ? values.definition.default : undefined,
+    },
+  };
+}
+
+export const InputsView: React.FC = () => {
+  const { formatMessage } = useIntl();
+  const [inputs, , helpers] = useField<BuilderFormInput[]>("inputs");
+  const [inputInEditing, setInputInEditing] = useState<InputInEditing | undefined>(undefined);
+  const usedKeys = useMemo(() => inputs.value.map((input) => input.key), [inputs.value]);
+  const inputInEditValidation = useMemo(
+    () =>
+      yup.object().shape({
+        // make sure key can only occur once
+        key: yup
+          .string()
+          .required("form.empty.error")
+          .notOneOf(inputInEditing?.isNew ? usedKeys : usedKeys.filter((key) => key !== inputInEditing?.key)),
+        required: yup.bool(),
+        definition: yup.object().shape({
+          title: yup.string().required("form.empty.error"),
+        }),
+      }),
+    [inputInEditing?.isNew, inputInEditing?.key, usedKeys]
+  );
+  return (
+    <BuilderConfigView heading={formatMessage({ id: "connectorBuilder.inputsTitle" })}>
+      <Text centered className={styles.inputsDescription}>
+        <FormattedMessage id="connectorBuilder.inputsDescription" />
+      </Text>
+      {inputs.value.length > 0 && (
+        <Card withPadding className={styles.inputsCard}>
+          <ol className={styles.list}>
+            {inputs.value.map((input) => (
+              <li className={styles.listItem} key={input.key}>
+                <div className={styles.itemLabel}>{input.definition.title || input.key}</div>
+                <Button
+                  className={styles.itemButton}
+                  size="sm"
+                  variant="secondary"
+                  aria-label="Edit"
+                  onClick={() => {
+                    setInputInEditing(formInputToInputInEditing(input));
+                  }}
+                >
+                  <FontAwesomeIcon className={styles.icon} icon={faGear} />
+                </Button>
+              </li>
+            ))}
+          </ol>
+        </Card>
+      )}
+      <Button
+        className={styles.addInputButton}
+        onClick={() => {
+          setInputInEditing(newInputInEditing());
+        }}
+        icon={<FontAwesomeIcon icon={faPlus} />}
+        iconPosition="left"
+        variant="secondary"
+      >
+        <FormattedMessage id="connectorBuilder.addInputButton" />
+      </Button>
+      {inputInEditing && (
+        <Formik
```

Comment:
What I did to fix these 
* Clear default on type change as you suggested. I added a `onChange` prop to the builder field to do so as an effect hook will do it in another render cycle so the form would still render in the broken state
* For the dropdown with `...` the issue was that the BuilderField would automatically select the first option if no value is set yet which is the wrong behavior IMHO (it looks like the first option is selected but this is not actually the case). So when you enable default values for an enum user input, it is not picking the first value, but instead you have to choose. The enum builder field is only used in one other place which is the http method picker and it always has a value by the default state of a stream.
* Errors about duplicate field IDs weren't shown right away because the field wasn't touched. I worked around this by auto-touching the key field and setting the title-derived key right away in the newly introduced onChange instead of waiting for a hook to trigger

Reply:
- Looks good
- That makes sense, I agree that it makes sense to make the user actually select a value if no default is set
- Nice, that seems to work well to me!